### ISSUE TITLE
Media manager not upload some files

### ISSUE THREAD

**Issue by taur-firiath** [joomla/joomla-cms#8067]
I can not upload files ~~over 16 MB~~.
In /var/log/httpd/error_log i have PHP Warning:  Invalid argument supplied for foreach() in /var/www/html/administrator/components/com_media/controllers/file.php on line 94 and line 131.

error_log( print_R($files,TRUE) ) logs temporary file name and etc. if file size less than 16MB, and nothing otherwise.

Joomla and PHP has file size limit at 32MB.

This occurs after ugrade to Joomla 3.4.4 from 3.3.6 through web interface updater.

[Joomla system information](https://github.com/taur-firiath/bug/blob/master/joomla/joomla-cms/8067/System%20Information.htm)

**Comment by taur-firiath** [comment]
What information should I give you?

**Comment by ggppdk** [comment]
This sounds like a server issue:
- maybe related to your PHP version / web server version
- maybe not enough disk space (not in your case)

Normally $_FILES  is not empty
- it is not empty even if file-size limits are exceeded

Joomla com_media code retrieves it and 
it checks it for errors, you should get a Joomla Error message:

Exceeded  PHP  upload_max_filesize 
- This file is too large to upload.

Exceeded  PHP  post_max_size
- Total size of upload exceeds the limit

**Comment by ggppdk** [comment]
Also test with (PHP) memory_limit  
32M or 50M

**Comment by taur-firiath** [comment]
In my current configuration memory_limit is 256MB. I tried 64MB and nothing changed.
On slow connection i can see that file is uploading (i wait), but nothing in media manager.
In file.php i also logs $contentLength and it is approximately equal to file size.

Today i try upload 3.5MB file and nothing happens. It is random thing (i change theme title). And i append my system information to first post.

Should i log anything else? Or enable debug mode?

**Comment by brianteeman** [comment]
This really sounds like a hosting issue

**Comment by joewebber** [comment]
I am also experiencing this issue after upgrade to 3.4.4

Certain files will upload fine, but I have a zip 2.5MB zip file that will not upload.
I have added some debugging to /administrator/components/com_media/controllers/file.php and the '$files' variable is empty.

I was able to get the script to read the data by changing this variable to get the data directly from the $_FILES array, but this fails with the error 'File is too large to upload' further along in the script.

**Comment by joomdonation** [comment]
@joewebber : Try to change the code from
```php
$files        = $this->input->files->get('Filedata', '', 'array');
```
To
```php
$files        = $this->input->files->get('Filedata', '', 'raw');
```
might solve the issue. I had to make the same change in my custom extension but haven't had a chance to check why the original code doesn't work yet.

**Comment by joewebber** [comment]
@joomdonation That works - thanks

**Comment by taur-firiath** [comment]
I change 'array' to 'raw' and get "File not uploaded for security reasons", but $files is not empty.
In my media settings 'zip', 'ZIP' extensions and 'application/x-zip' mime already has been allowed. I append 'application/x-zip-compressed' - fail. Then i disable 'Restrict Uploads' and 'Check MIME Types' - fail.

Then i find [post](https://compojoom.com/forum/134-jedchecker/20818-not-uploaded-for-security-reasons#32296) and change JFile::upload() in file.php. File uploading works now.

**Comment by roland-d** [comment]
@taur-firiath are you uploading any files with PHP code inside them? The post you found disables the security check, that is why it works for you.

**Comment by taur-firiath** [comment]
> files with PHP code inside

No. Images and PDFs, for example.

[3.5MB PDFs](https://github.com/taur-firiath/bug/blob/master/joomla/joomla-cms/8067/test.zip) NOT uploads without above hacks.
[16MB images](https://github.com/taur-firiath/bug/blob/master/joomla/joomla-cms/8067/Desktop16.zip) NOT uploads without above hacks.
[8MB images](https://github.com/taur-firiath/bug/blob/master/joomla/joomla-cms/8067/Desktop8.zip) uploads with or without hacs.

**Comment by roland-d** [comment]
@taur-firiath I have uploaded the 3.5MBs you provided without a problem, first as all 6 single files at once and later as one big PDF, no issue.

The images were no problem also but it is a combination of factors. You need to have the memory_limit, post_max_size, upload_max_filesize all set large enough for the upload to handle it.

I am closing this issue as it is not an issue with Joomla. Thank you for your contribution. If you need further help on this please check with on the http://forum.joomla.org

The post you found to work around the upload restriction is just that a workaround. Joomla does scan uploaded files for possible hidden PHP code.

**Comment by wesley-if** [comment]
Actually I'm facing this problem too. I have 2 zip files which contains only several pdf files, file size is about 10.xMB each.
$_FILES is not empty but $this->input->files return me empty value.

Initially I thought it was caused by file upload limit or post limit set in php.ini. It failed to upload on my localhost too where the limit already set to 128MB.

Finally I found out JFilterInput::isSafeFile return false when my zip file data (fread) contains the value in 'forbidden_extensions' which is 'php', 'phps', 'php5', 'php3', 'php4', 'inc', 'pl', 'cgi', 'fcgi', 'java', 'jar', 'py'.

Running Joomla! 3.4.8.
/libraries/joomla/filter/input.php - return false at line 552
